Formula/o/opencascade.rb

```diff
@@ -72,6 +72,35 @@ def install
     system "cmake", "--build", "build"
     system "cmake", "--install", "build"
 
+    # The soname / install name of libtbb and libtbbmalloc are versioned only
+    # by the minor version (e.g., `libtbb.so.12`), but Open CASCADE's CMake
+    # config files reference the fully-versioned filenames (e.g.,
+    # `libtbb.so.12.11`).
+    # This mandates rebuilding opencascade upon tbb's minor version updates.
+    # To avoid this, we change the fully-versioned references to the minor-only
+    # version. For example:
+    #   libtbb.so.12.11 => libtbb.so.12
+    #   libtbbmalloc.so.2.11 => libtbbmalloc.so.2
+    #   libtbb.12.11.dylib => libtbb.12.dylib
+    #   libtbbmalloc.2.11.dylib => libtbbmalloc.2.dylib
+    # See also:
+    #   https://github.com/Homebrew/homebrew-core/issues/129111
+    #   https://dev.opencascade.org/content/cmake-files-macos-link-non-existent-libtbb128dylib
+    tbb_regex = /
+      libtbb
+      (?<malloc>malloc)?
+      (?<so>\.so)?
+      \.(?<major>\d+)
+      \.(?<minor>\d+)
+      (?<dylib>\.dylib)?
+    /x
+    inreplace (lib/"cmake/opencascade").glob("*.cmake") do |s|
+      s.gsub! tbb_regex do
+        match = Regexp.last_match
+        "libtbb#{match[:malloc]}#{match[:so]}.#{match[:major]}#{match[:dylib]}"
+      end
+    end
```

Comment:
@carlocab: Since you handled this previously (in #129111 and #129467), maybe you'd like to have a look at this.